x-pack/plugins/transform/public/app/common/request.ts

```diff
@@ -57,17 +57,45 @@ export function isDefaultQuery(query: PivotQuery): boolean {
   return isSimpleQuery(query) && query.query_string.query === '*';
 }
 
+export function getCombinedRuntimeMappings(
+  indexPattern: IndexPattern | undefined,
+  runtimeMappings?: StepDefineExposedState['runtimeMappings']
+): StepDefineExposedState['runtimeMappings'] | undefined {
+  let combinedRuntimeMappings = {};
+
+  // Use runtime field mappings defined inline from API
+  if (typeof runtimeMappings === 'object') {
+    combinedRuntimeMappings = { ...combinedRuntimeMappings, ...runtimeMappings };
+  }
+
+  // And runtime field mappings defined by index pattern
+  if (indexPattern !== undefined) {
+    const ipRuntimeMappings = indexPattern.getComputedFields().runtimeFields;
+    combinedRuntimeMappings = { ...combinedRuntimeMappings, ...ipRuntimeMappings };
+  }
+
+  if (
+    typeof combinedRuntimeMappings === 'object' &&
+    Object.keys(combinedRuntimeMappings).length > 0
```

Comment:
is it worth checking for `null`?

Reply:
Updated here https://github.com/elastic/kibana/commit/bc435134b7a10eb60657c077bec7a52659473569